### PR TITLE
Enhancements to cci project init

### PR DESCRIPTION
**Pull request by jlantz**
I had previously left `cci project init` alone in favor of using a cookiecutter template to launch the project config.  The problem is that we don't have as much control over the process with cookiecutter, cookiecutter is another dependency, and cookiecutter can't run against an existing directory.

The two most common use cases we really want to handle are:
1. You're starting a new project
2. You have a Github repo with metadata under `src/` and you want to enable that repo for CumulusCI

Cookiecutter won't help in use case #2.  This new task will run against any local git repository that doesn't already have a `cumulusci.yml` file and configures the following:

* `cumulusci.yml`
* `sfdx-project.json` if none exists already, otherwise leaves existing file alone
* `orgs/*.json` Starter files for defining org shapes

### CODE REVIEW

File: cumulusci/cli/cli.py [hunk 1]
```diff
@@ -305,115 +317,153 @@ def service(config):
...
-        package_config.append(
-            '        namespace: {}'.format(package_namespace))
+
+    click.echo()
+    package_namespace = None
+    if click.confirm(click.style("Is this a managed package project?", bold=True), default=False):
+        click.echo('Enter the namespace assigned to the managed package for this project')
+        package_name = click.prompt(click.style('Package Namespace', bold=True), default=name)
+        if package_namespace and package_namespace != config.global_config.project__package__namespace:
+            package_config.append(
```
Comment (cdcarter) [inline cumulusci/cli/cli.py]: I have to ask if theres a way to build this config in memory as a dict and just dump to yaml?

Comment (cdcarter) [reply cumulusci/cli/cli.py]: i know the yaml strings are't new but...

File: cumulusci/cli/cli.py [hunk 2]
```diff
@@ -305,115 +317,153 @@ def service(config):
...
+        "namespace": package_namespace,
+        "sourceApiVersion": package_api_version,
+}
+    with open('sfdx-project.json','w') as f:
+        f.write(json.dumps(sfdx_project))
+
+    if not os.path.isdir('orgs'):
+        os.mkdir('orgs')
+      
+    org_content_url = 'https://raw.githubusercontent.com/SalesforceFoundation/sfdo-package-cookiecutter/master/%7B%7Bcookiecutter.project_name%7D%7D/orgs/{}.json'
```
Comment (cdcarter) [inline cumulusci/cli/cli.py]: this...worries me...

I know it was simple but maybe lets just include these files in cci distribution?

File: cumulusci/cumulusci.yml [hunk 3]
```diff
@@ -462,8 +462,20 @@ services:
...
+        required: False
+      url:
+        description: The main url to the MetaCI site
+        required: True
+      token:
+        description: Your API token to the MetaCI site (get from SITE_URL/api/token)
+        required: True
   mrbelvedere:
-    description: Configure connection for mrbelvedere tasks, e.g. MrbelvederePublish
+    descript: Configure connection for mrbelvedere tasks, e.g. MrbelvederePublish
```
Comment (cdcarter) [inline cumulusci/cumulusci.yml]: description? descript?

Comment (jlantz) [reply cumulusci/cumulusci.yml]: Oops, fixed

File: cumulusci/cli/cli.py (hunk 1 above)
Comment (jlantz) [reply cumulusci/cli/cli.py]: The yaml dump doesn't respect ordereddict and even the ordereddict dumper in HiYaPyCo which we have available doesn't handle recursive dictionaries very well.  I do want to revisit this at some point, probably with building a model of some sort for the cci config and then being able to render that as json, yaml, etc.   We could use that model and its validation in a bunch of places.

File: cumulusci/cli/cli.py (hunk 2 above)
Comment (jlantz) [reply cumulusci/cli/cli.py]: Probably something we should do eventually but can be done in a future commit.  This definitely works for now at least.

File: cumulusci/cli/cli.py (hunk 1 above)
Comment (cdcarter) [reply cumulusci/cli/cli.py]: marshmallow!

**Pull request merged**